src/analysis/retail/paladin/holy/modules/beacons/BeaconTargets.ts

```diff
@@ -20,11 +19,15 @@ class BeaconTargets extends Analyzer {
   protected combatants!: Combatants;
 
   currentBeaconTargets: number[] = [];
+  currentBeaconTargetsByBeaconId: { [beaconId: number]: number[] } = {};
   maxBeacons = 1;
 
   hasBeacon(playerId: number) {
     return this.currentBeaconTargets.includes(playerId);
   }
+  specificBeaconTargets(beaconId: number) {
+    return (this.currentBeaconTargetsByBeaconId[beaconId] ?? []).length;
+  }
```

Comment:
You may want to rename this to something like `getBeaconTargetForId` or something similar